exec/compile_test.go

```diff
@@ -285,3 +366,45 @@ func lineDiff(lhs, rhs string) string {
 	}
 	return b.String()
 }
+
+type fakeShardCache struct {
```

Comment:
Optional: Can we use the real cache slice here, but override the cache field? I think it'd be ok to export it and document that it's for internal use (or implement some other hiding mechanism).

Reply:
When we do that, we have this issue:

The call to construct the slice, `Cache`, creates the shard cache.  Creating the shard cache checks the filesystem for cache files.  We could do something like use a gibberish (or known empty temp) prefix to make sure that nothing untoward happens as a result, but I didn't want to rely too much on the shard cache internals in this compilation test.